#! /bin/sh /usr/share/dpatch/dpatch-run
## 08-skip-bad-utf-on-pdf-import.dpatch by Thomas Viehmann <tv@beamnet.de> 
##
## DP: This patch closes bug #510234.
## DP: Thomas' report and description of the patch:
## DP: When converting the attached testcase.pdf (generated by latex testcase ;
## DP: dvips testcase ; ps2pdf testcase.ps (*not* pdflatex) from the tex),
## DP: inkscape currently fails and with the patch works.
## DP: There seem to be several exceptions to catch here, I observed
## DP: std:length_error (because string::append was passed length
## DP: string::n_pos) and possibly Glib::ConvertError.

@DPATCH@

--- inkscape-0.46/src/extension/internal/pdfinput/svg-builder.cpp~
+++ inkscape-0.46/src/extension/internal/pdfinput/svg-builder.cpp
@@ -1208,11 +1208,16 @@
         os_y << delta_pos[1];
         y_coords.append(os_y.str());
         last_delta_pos = delta_pos;
-
-        // Append the character to the text buffer
-        text_buffer.append((char *)&glyph.code, 1);
-
-        glyphs_in_a_row++;
+        
+        try {
+            // Append the character to the text buffer
+            text_buffer.append((char *)&glyph.code, 1);
+            glyphs_in_a_row++;
+        }
+        catch (std::exception& e) {
+            std::cerr << "Error converting glyphs in PDF: " << e.what()<< std::endl; 
+        }
+        
         i++;
     }
     _container->appendChild(text_node);
